sdk/storage/inc/datalake/file_system_client.hpp

```diff
@@ -0,0 +1,124 @@
+     * @param credential The shared key credential used to initialize the client.
+     * @param options Optional parameters used to initialize the client.
+     */
+    explicit FileSystemClient(
+        const std::string& fileSystemUri,
+        std::shared_ptr<SharedKeyCredential> credential,
+        const FileSystemClientOptions& options = FileSystemClientOptions());
+
+    /**
+     * @brief Bearer token authentication client.
+     * @param fileSystemUri The URI of the file system this client's request targets.
+     * @param credential The token credential used to initialize the client.
+     * @param options Optional parameters used to initialize the client.
+     */
+    explicit FileSystemClient(
+        const std::string& fileSystemUri,
+        std::shared_ptr<TokenCredential> credential,
+        const FileSystemClientOptions& options = FileSystemClientOptions());
+
+    /**
+     * @brief Anonymous/SAS/customized pipeline auth.
+     * @param fileSystemUri The URI of the file system this client's request targets.
+     * @param options Optional parameters used to initialize the client.
+     */
+    explicit FileSystemClient(
+        const std::string& fileSystemUri,
+        const FileSystemClientOptions& options = FileSystemClientOptions());
+
+    /**
+     * @brief Create a PathClient from current FileSystemClient
+     * @param path Path of the resource within the file system.
+     * @return PathClient
+     */
+     PathClient GetPathClient(const std::string& path) const;
+
+    /**
+     * @brief Creates the file system.
+     * @param options Optional parameters to create this file system.
+     * @return FileSystemCreateResponse
+    */
+    FileSystemCreateResponse Create(
+        const CreateFileSystemOptions& options = CreateFileSystemOptions()) const;
+
+    /**
+     * @brief Deletes the file system.
+     * @param options Optional parameters to delete this file system.
+     * @return FileSystemDeleteResponse
+     */
+    FileSystemDeleteResponse Delete(
+        const DeleteFileSystemOptions& options = DeleteFileSystemOptions()) const;
+
+    /**
+     * @brief Gets the properties of file system.
+     * @param options Optional parameters to get the properties of this file system.
+     * @return FileSystemGetPropertiesResponse
+     * @remark You can leverage ParseDataLakePropertiesString to parse the properties string.
+     */
+    FileSystemGetPropertiesResponse GetProperties(
+        const GetFileSystemPropertiesOptions& options = GetFileSystemPropertiesOptions()) const;
+
+    /**
+     * @brief Sets the properties of file system.
+     * @param options Optional parameters to set the properties to this file system.
+     * @return FileSystemSetPropertiesResponse
+     */
+    FileSystemSetPropertiesResponse SetProperties(
+        const SetFileSystemPropertiesOptions& options = SetFileSystemPropertiesOptions()) const;
```

Comment:
I think there should be a mandatory parameter here

Reply:
The `properties` is still optional here according to the [official doc](https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/filesystem/setproperties).

Reply:
This is called `SetHttpHeaders` in other SDK.

Reply:
Changed to `SetMetadata` already.